e2e/src/auction_bid.rs

```diff
@@ -0,0 +1,87 @@
+use ethcontract::web3::types::{H160, H256, U256};
+
+#[derive(Debug)]
+pub struct AuctionBid {
+    order_hash: H256,
+    num_orders: u64,
+    creation_time_stamp: U256,
+    pub solver: H160,
+    objective_value: U256,
+    pub tentative_state: H256,
+    solution_hash: H256,
+}
+
+type PreAuctionBid = (
```

Comment:
Maybe a doc comment explaining why this type is useful. When I first read this I was confused about it until I saw `AuctionBid::from(&instance.auctions(auction_index).wait_and_expect(".."))`.

Reply:
Great idea. Will do I am still not entirely satisfied with this `AuctionBid` model yet and may even consider removing it completely.

Reply:
Renamed this type to `AuctionBidData`